### PR TITLE
Telemetry gaps downloads

### PR DESCRIPTION
**Pull request by jonathanmendez**
Add file path and file hash to download telemetry for enterprise.

Also fix download_attempted telemetry to obey url/file telemetry prefs and use same 'source_url' key.

### CODE REVIEW

File: browser/components/downloads/DownloadsTelemetry.enterprise.sys.mjs [hunk 1]
```diff
@@ -221,8 +231,10 @@ export const DownloadsTelemetryEnterprise = {
 
       const telemetryData = {
         filename: fileInfo.filename,
+        file_path: fileInfo.file_path,
         extension: fileInfo.extension,
         mime_type: fileInfo.mime_type,
+        sha256_hash: download.saver.getSha256Hash(),
```
Comment (gcp) [inline browser/components/downloads/DownloadsTelemetry.enterprise.sys.mjs]: Ok that was easy :)

File: toolkit/components/downloads/DownloadCore.sys.mjs [hunk 2]
```diff
@@ -720,23 +831,24 @@ Download.prototype = {
...
       Glean.downloads.downloadAttempt.record({
-        filename,
-        extension: fileExtension,
-        mime_type: this.contentType || "",
-        source_url_domain: sourceDomain,
+        filename: fileInfo.filename,
+        file_path: fileInfo.file_path,
+        extension: fileInfo.extension,
+        mime_type: fileInfo.mime_type,
+        source_url: sourceUrl,
```
Comment (gcp) [inline toolkit/components/downloads/DownloadCore.sys.mjs]: You wrote `processSourceUrl` so that this can be `null`. Does that work for Glean? (I would suspect not because all other stuff is defaulted to `""`?)

File: browser/components/downloads/DownloadsTelemetry.enterprise.sys.mjs (hunk 1 above)
Comment (gcp) [reply browser/components/downloads/DownloadsTelemetry.enterprise.sys.mjs]: Is this guaranteed to exist? (If not, same remark as with `source_url` below)

From digging, one case would be external protocol handlers, but then we also don't really download anything, so I guess that's fine?

**Review by gcp — CHANGES_REQUESTED** [review]
Probably can't send `null` in the indicated case.

File: toolkit/components/downloads/DownloadCore.sys.mjs (hunk 2 above)
Comment (jonathanmendez) [reply toolkit/components/downloads/DownloadCore.sys.mjs]: Good catch, forgot to copy the default over from the downloadcompleted code.

File: browser/components/downloads/DownloadsTelemetry.enterprise.sys.mjs (hunk 1 above)
Comment (jonathanmendez) [reply browser/components/downloads/DownloadsTelemetry.enterprise.sys.mjs]: From my own digging, it seemed like if we did reach the success case it should exist? But it doesn't hurt to check against null.

FWIW, since I noticed that sizeBytes could also technically be null, I went ahead and tested what happens if you pass null; the event still got sent and received (at least by my local http-edge), and the event was just missing that key from the data. That being said, I'll go ahead and default it to 0 just in case.

File: browser/components/downloads/DownloadsTelemetry.enterprise.sys.mjs [hunk 3]
```diff
@@ -221,9 +231,11 @@ export const DownloadsTelemetryEnterprise = {
 
       const telemetryData = {
         filename: fileInfo.filename,
+        file_path: fileInfo.file_path,
         extension: fileInfo.extension,
         mime_type: fileInfo.mime_type,
-        size_bytes: sizeBytes,
+        sha256_hash: download.saver.getSha256Hash() || "",
+        size_bytes: sizeBytes || 0,
```
Comment (gcp) [inline browser/components/downloads/DownloadsTelemetry.enterprise.sys.mjs]: Why not fix the code right above? (...which sets `size_bytes` to null to begin with!)

**Review by gcp — APPROVED** [review]
OK to land but one more cleanup I'd like to see.

File: browser/components/downloads/DownloadsTelemetry.enterprise.sys.mjs (hunk 3 above)
Comment (jonathanmendez) [reply browser/components/downloads/DownloadsTelemetry.enterprise.sys.mjs]: Done!
